### DISCUSSION TITLE
Merging Temporary Session Cart with User Database Cart

### DISCUSSION THREAD

**Discussion by ErickaVea** [pimcore/pimcore#16424] [Q&A / Support] [answered]
Hello everyone,

I'm working on a project using the Pimcore E-Commerce Framework and have encountered a challenge. I need to merge a temporary session cart (for non-registered users) with a registered user's cart stored in the database. I've set up the carts as shown in the following code snippet:

![cart](https://github.com/pimcore/pimcore/assets/100506118/80958c9b-91f5-460a-8bb3-39a857a91ee5)

Additionally, I have implemented the getCart() and mergeTemporaryCartIntoUserCart() methods to handle the cart logic. Here's the relevant code:

![getCart](https://github.com/pimcore/pimcore/assets/100506118/7cd29ebc-4286-47af-addb-54704bcd7814)

My question is: What would be the best approach to merge these two carts (the temporary session cart and the user cart from the database) to ensure the transition is smooth and no data is lost? Any suggestions or code examples would be greatly appreciated.

Thank you in advance for your help.
@weisswurstkanone @dvesh3 @brusch @NiklasBr @mcop1

**Comment by fashxp** [comment, marked as answer]
maybe that helps you a bit? https://github.com/pimcore/demo/blob/11.x/src/EventListener/AuthenticationLoginListener.php#L80

**Reply by ErickaVea** [reply]
Thank you very much, it helped me a lot!
